pkg/operator/configobservation/node/observe_minimum_kubelet_version.go

```diff
@@ -61,13 +71,63 @@ func (o *minimumKubeletVersionObserver) ObserveMinimumKubeletVersion(genericList
 		// return empty set of configs, this helps to unset the config
 		// values related to the minimumKubeletVersion.
 		// Also, ensures that this observer doesn't break cluster upgrades/downgrades
-		return map[string]interface{}{}, errs
+		return ret, errs
 	}
 
-	ret = map[string]interface{}{}
 	if err := unstructured.SetNestedField(ret, configNode.Spec.MinimumKubeletVersion, minimumKubeletVersionConfigPath); err != nil {
-		errs = append(errs, err)
+		return ret, append(errs, err)
 	}
 
 	return ret, errs
 }
+
+type authorizationModeObserver struct {
+	featureGateAccessor featuregates.FeatureGateAccess
+}
+
+func NewAuthorizationModeObserver(featureGateAccessor featuregates.FeatureGateAccess) configobserver.ObserveConfigFunc {
+	return (&authorizationModeObserver{
+		featureGateAccessor: featureGateAccessor,
+	}).ObserveAuthorizationMode
+}
+
+// ObserveAuthorizationMode watches the featuregate configuration and generates the apiServerArguments.authorization-mode
+// It currently hardcodes the default set and adds MinimumKubeletVersion if the feature is set to on.
+func (o *authorizationModeObserver) ObserveAuthorizationMode(genericListers configobserver.Listers, _ events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	ret = map[string]interface{}{}
+	if !o.featureGateAccessor.AreInitialFeatureGatesObserved() {
+		return existingConfig, nil
+	}
+
+	featureGates, err := o.featureGateAccessor.CurrentFeatureGates()
+	if err != nil {
+		return existingConfig, append(errs, err)
+	}
+
+	defer func() {
+		// Prune the observed config so that it only contains minimumKubeletVersion field.
+		ret = configobserver.Pruned(ret, authModePath)
+	}()
+
+	if err := SetAPIServerArgumentsToEnforceMinimumKubeletVersion(existingConfig, !featureGates.Enabled(features.FeatureGateMinimumKubeletVersion)); err != nil {
+		return existingConfig, append(errs, err)
+	}
+	return existingConfig, nil
+}
+
+// SetAPIServerArgumentsToEnforceMinimumKubeletVersion modifies the passed in config
+// to add the "authorization-mode": "MinimumKubeletVersion" if the feature is on. If it's off, it
+// removes it instead.
+func SetAPIServerArgumentsToEnforceMinimumKubeletVersion(existingConfig map[string]interface{}, on bool) error {
```

Comment:
Does this need the existing config at all? You could remove that and skip the pruning since this is roughly `if on { return append(defaults, another) } else { return defaults }`.